direct/src/gui/DirectFrame.py

```diff
@@ -50,166 +52,136 @@ def __init__(self, parent = None, **kw):
             # Change default value of text mayChange flag from 0
             # (OnscreenText.py) to 1
             ('textMayChange',  1,          None),
-            )
+        )
         # Merge keyword options with default options
         self.defineoptions(kw, optiondefs,
-                           dynamicGroups = DirectFrame.DefDynGroups)
+                           dynamicGroups=DirectFrame.DefDynGroups)
 
         # Initialize superclasses
         DirectGuiWidget.__init__(self, parent)
 
         # Call option initialization functions
         self.initialiseoptions(DirectFrame)
 
-    def destroy(self):
-        DirectGuiWidget.destroy(self)
-
     def clearText(self):
         self['text'] = None
         self.setText()
 
     def setText(self, text=None):
         if text is not None:
-            self['text'] = text
-
-        # Determine if user passed in single string or a sequence
-        if self['text'] == None:
-            textList = (None,) * self['numStates']
-        elif isinstance(self['text'], stringType):
-            # If just passing in a single string, make a tuple out of it
-            textList = (self['text'],) * self['numStates']
+            self["text"] = text
+
+        text = self["text"]
+        component_class = OnscreenText
+        component_kwargs ={
+                "scale": 1,
+                "mayChange": self['textMayChange'],
+                "sort": DGG.TEXT_SORT_INDEX, # this is "30"
+        }
```

Comment:
This dict has spacing issues; but then, I think it would be cleaner to just move this down as keyword arguments to the reinitComponent call.

Reply:
So, reninitComponent(scale=1)? I disagree.

The point of all this in the first place is that createcomponent is superfluous. Everything that setImage and now __reinitComponent and createcomponent do, is collect inputs to call the init's of Onscreen[Stuff].

That is where these defaults belong.

But I can't replace createcomponent in one go, so collecting inputs like this will make the next PR where I do remove create component more clear. 

Then I can move the default values to the inits and just remove the dict here.